src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/oops/Oop.java

```diff
@@ -75,8 +80,17 @@ private static synchronized void initialize(TypeDataBase db) throws WrongTypeExc
 
   // Accessors for declared fields
   public Mark  getMark()   { return new Mark(getHandle()); }
+
+  private static Klass getKlass(Mark mark) {
+    assert(VM.getVM().isCompactObjectHeadersEnabled());
```

Comment:
`mark.getKlass()` already does this assert. I don't see any value in this `getKlass()` method. The caller should just call `getMark().getKlass()` rather than `getKlass(getMark())`.

Reply:
I'm not sure why this got marked as resolved.